src/sss_client/pam_sss.c

```diff
@@ -1776,6 +1790,121 @@ static int prompt_oauth2(pam_handle_t *pamh, struct pam_items *pi)
     return PAM_SUCCESS;
 }
 
+static int prompt_passkey(pam_handle_t *pamh, struct pam_items *pi,
+                          const char *prompt_interactive, const char *prompt_touch)
+{
+    int ret;
+    const struct pam_conv *conv;
+    const struct pam_message *mesg[3] = { NULL, NULL, NULL };
+    struct pam_message m[3] = { {0}, {0}, {0} };
+    struct pam_response *resp = NULL;
+    int pin_idx = 0;
+    int msg_idx = 0;
+
+    ret = pam_get_item(pamh, PAM_CONV, (const void **) &conv);
+    if (ret != PAM_SUCCESS) {
+        return ret;
+    }
+    if (conv == NULL || conv->conv == NULL) {
+        logger(pamh, LOG_ERR, "No conversation function");
+        return PAM_SYSTEM_ERR;
+    }
+
+    pi->pam_authtok_type = SSS_AUTHTOK_TYPE_OAUTH2;
+	/* Interactive, prompt a message and wait before continuing */
+    if ((strcmp(prompt_interactive, "")) != 0) {
+	    m[msg_idx].msg_style = PAM_PROMPT_ECHO_OFF;
+	    m[msg_idx].msg = prompt_interactive;
+	    msg_idx++;
+    }
+
+    /* Prompt for PIN
+     *
+     * If prompt_pin is false but a PIN is set on the device
+     * we still prompt for PIN */
+    if ((strcmp(pi->passkey_prompt_pin, "true")) == 0) {
+        m[msg_idx].msg_style = PAM_PROMPT_ECHO_OFF;
+        m[msg_idx].msg = PASSKEY_DEFAULT_PIN_MSG;
+        pin_idx = msg_idx;
+        msg_idx++;
+    }
+
+    /* Prompt to remind the user to touch the device */
+    if ((strcmp(prompt_touch, "") != 0)) {
```

Comment:
same here -- if you want to check that the string is not empty, `(foo != NULL && foo[0] != '\0')` is better than `strcmp()`

Reply:
Fixed.